Validation/RecoTau/src/TauTagValidation.cc

```diff
@@ -367,6 +383,15 @@ void TauTagValidation::analyze(const edm::Event& iEvent, const edm::EventSetup&
     tauDeacyCountMap_.insert(std::make_pair("twoProng2Pi0" + DiscriminatorLabel, 0.));
     tauDeacyCountMap_.insert(std::make_pair("threeProng0Pi0" + DiscriminatorLabel, 0.));
     tauDeacyCountMap_.insert(std::make_pair("threeProng1Pi0" + DiscriminatorLabel, 0.));
+    tauDeacyCountMap_.find( "allHadronic"   + DiscriminatorLabel)->second = 0;
+    tauDeacyCountMap_.find( "oneProng0Pi0"  + DiscriminatorLabel)->second = 0;
+    tauDeacyCountMap_.find( "oneProng1Pi0"  + DiscriminatorLabel)->second = 0;
+    tauDeacyCountMap_.find( "oneProng2Pi0"  + DiscriminatorLabel)->second = 0;
+    tauDeacyCountMap_.find( "twoProng0Pi0"  + DiscriminatorLabel)->second = 0;
+    tauDeacyCountMap_.find( "twoProng1Pi0"  + DiscriminatorLabel)->second = 0;
+    tauDeacyCountMap_.find( "twoProng2Pi0"  + DiscriminatorLabel)->second = 0;
+    tauDeacyCountMap_.find( "threeProng0Pi0"+ DiscriminatorLabel)->second = 0;
+    tauDeacyCountMap_.find( "threeProng1Pi0"+ DiscriminatorLabel)->second = 0;
```

Comment:
Why you set values for these keys (to 0)? It looks that they are already set when `insert`'ed a few lines above?

Reply:
Dennis/Markus noticed that the tau count is not initialized. In fact you have histograms where the avarage is O(1000), meaning that in each event the number of taus is not initialized at zero. With this change, the histograms look reasonable.

Reply:
'0.' used in `make_pair` is not an int, please try with '0'. Further value settings should be unnecessary, please verify it e.g. with printouts.